server/container/util-java8/src/main/java/org/apache/james/util/OptionalUtils.java

```diff
@@ -45,4 +45,17 @@
             .filter(Optional::isPresent)
             .orElse(optional2);
     }
+
+    public static <T> boolean containsDifferent(Optional<T> requestValue, T storeValue) {
```

Comment:
I'm against having such method : it's barely shorter then before and it's not simpler to read IMO.

Reply:
I do not agree, the name is very clear and it lead to shorter code IMO.

Reply:
Yes, because this function is called twice at least so it makes sense to extract to util method